src/olympia/files/tasks.py

```diff
@@ -57,13 +58,18 @@ def repack_fileupload(results, upload_pk):
     # developer, so it could be something else.
     if upload.path.endswith('.xpi'):
         try:
-            tempdir = extract_zip(upload.path)
-        except Exception:
+            tempdir = tempfile.mkdtemp()  # *not* on TMP_PATH, we want local fs
+            zip_file = SafeZip(source=upload.path)
+            zip_file.extract_to_dest(tempdir)
```

Comment:
I know it does not have anything to do with this patch but it'd be nice to track the timing of this task, or even better, to track:

- the extraction time
- the repacking time
- the time to move the file back

Reply:
We do log start/end timings for all tasks already, but maybe having more granular timings would be nice, yes. I'll look into that.

Reply:
d35c1670e30a8c55b24310b4f4e3bd674fd76c17 adds individual timings